tools-local/doc.tasks/GenerateMarkdownDoc.cs

```diff
@@ -94,15 +96,31 @@ private void WriteSectionProjecs(ITaskItem section, StreamWriter file)
             file.WriteLine($"Projects in section: {projectFiles.Length}\n");
                     
             Array.Sort(projectFiles, (a, b) => a.CompareTo(b));
+            Array.ForEach(projectFiles, pf => pf = RelativizePath(pf));
 
-            for (int i = 0; i < projectFiles.Length; i++)
+            var groupedPaths = projectFiles
+                .Where(pf => pf.StartsWith(RepositoryRoot))
+                .Select(pf => pf.Substring(pf.IndexOf(RepositoryRoot) + RepositoryRoot.Length))
+                .Select(pf => pf.Split('\\'))
```

Comment:
Try to generalize the solution for paths of any length. Possible algorithm would be:

0. Get relative paths

  - "a\b\c\d\d,csproj"
  - "a\b\c\e\e,csproj"
  - "a\c\x\x,csproj"

1. Get the file name  

  - ("a\b\c\d\d,csproj", "a")
  - ("a\b\c\e\e,csproj", "e")
  - ("a\c\x\x,csproj", "x")

2. Remove the filename and the lastthe folder with the same name as the filename

  - ("a\b\c\d\d,csproj", "a", "a\b\c")
  - ("a\b\c\e\e,csproj", "e", "a\b\c")
  - ("a\c\x\x,csproj", "x", "a\b")

3. Group by the last tuple elment:

  - "a\b\c" => ("a\b\c\d\d,csproj", "a", "a\b\c") , ("a\b\c\e\e,csproj", "e", "a\b\c")
  - "a\b" => ("a\c\x\x,csproj", "x", "a\b")

Reply:
Not sure I'm quite following all of your examples, but I'll refactor to be a bit more generic. To be clear, you're saying we don't have these patterns today, but we might at some point?

Reply:
Yes. It's possible to have different folder patterns in the future. Happy to discuss the algorithm offline if needed.

Reply:
I believe I have addressed this. Please confirm